protocol/state/snapshot.go

```diff
@@ -1,53 +1,110 @@
-// Package state provides types for encapsulating blockchain state.
 package state
 
 import (
+	"fmt"
+
+	"chain/errors"
 	"chain/protocol/bc"
 	"chain/protocol/patricia"
 )
 
 // Snapshot encompasses a snapshot of entire blockchain state. It
-// consists of a patricia state tree and the issuances memory.
+// consists of a patricia state tree and the nonce set.
 //
-// Issuances maps an "issuance hash" to the time (in Unix millis)
-// at which it should expire from the issuance memory.
+// Nonces maps a nonce entry's ID to the time (in Unix millis)
+// at which it should expire from the nonce set.
 type Snapshot struct {
-	Tree      *patricia.Tree
-	Issuances map[bc.Hash]uint64
+	Tree   *patricia.Tree
+	Nonces map[bc.Hash]uint64
+}
+
+func (s *Snapshot) AddNonce(id bc.Hash, expiryMS uint64) error {
+	if s.Nonces[id] >= expiryMS {
+		return fmt.Errorf("conflicting nonce %x", id[:])
+	}
+	s.Nonces[id] = expiryMS
+	return nil
 }
 
-// PruneIssuances modifies a Snapshot, removing all issuance hashes
+func (s *Snapshot) DeleteSpentOutput(id bc.Hash) error {
+	if !s.Tree.Contains(id[:]) {
+		return fmt.Errorf("invalid prevout %x", id[:])
+	}
+	s.Tree.Delete(id[:])
+	return nil
+}
+
+func (s *Snapshot) AddOutput(id bc.Hash) error {
```

Comment:
Done. This (and two other functions, also now inlined) were there to satisfy a short-lived interface called `BlockchainState` used by `protocol/bc` when `Apply` functionality briefly lived there.